### PR TITLE
GitHub api limit

### PR DESCRIPTION
**Pull request by rekhaswaminathan**

### CODE REVIEW

File: src/Diagnostics.RuntimeHost/Services/GithubClient/GithubClient.cs [hunk 1]
```diff
@@ -195,6 +201,24 @@ public void Dispose()
             }
         }
 
+        public async Task<string> GetLatestSha()
+        {         
+            var branchInfo = await _octokitClient.Repository.Branch.Get(owner: _userName, name: _repoName, branch: _branch);
+            return branchInfo !=null && branchInfo.Commit != null ?  branchInfo.Commit.Sha : string.Empty;
+
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/GithubClient/GithubClient.cs:208]: nit: remove extra line

File: src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs [hunk 2]
```diff
@@ -151,8 +157,14 @@ private async Task StartWatcherInternal(bool startup)
...
                 var destLastModifiedMarker = await FileHelper.GetFileContentAsync(destDirInfo.FullName, _lastModifiedMarkerName);
+                
+                if(string.IsNullOrWhiteSpace(LatestSha))
+                {
+                    LatestSha = await _githubClient.GetLatestSha();
+                }
+
+                var response = await _githubClient.GetTreeBySha(sha: LatestSha, etag: destLastModifiedMarker);
 
-                var response = await _githubClient.Get(_rootContentApiPath, etag: destLastModifiedMarker);
```
Comment (hforeste) [inline src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs:155]: So when we make this call, this is getting multiple shas?

Comment (rekhaswaminathan) [reply src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs:155]: Nope this gets the full content of the github repo without truncating. 
This is the limitation of old API: https://developer.github.com/v3/repos/contents/#get-repository-content

Comment (hforeste) [reply src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs:155]: So now with this we don't have to keep manually removing some detectors. Nice.

**Pull request merged**